man/systemd.mount.xml

```diff
@@ -135,15 +135,17 @@
 
         <listitem><para>Mount units referring to local file systems automatically gain
         an <varname>After=</varname> dependency on <filename>local-fs-pre.target</filename>, and a
-        <varname>Before=</varname> dependency on <filename>local-fs.target</filename> unless
-        <option>nofail</option> mount option is set.</para></listitem>
-
-        <listitem><para>Network mount units
-        automatically acquire <varname>After=</varname> dependencies on <filename>remote-fs-pre.target</filename>,
-        <filename>network.target</filename> and <filename>network-online.target</filename>, and gain a
-        <varname>Before=</varname> dependency on <filename>remote-fs.target</filename> unless
-        <option>nofail</option> mount option is set. Towards the latter a
-        <varname>Wants=</varname> unit is added as well.</para></listitem>
+        <varname>Before=</varname> dependency on <filename>local-fs.target</filename> unless one or more
+        mount options among <option>nofail</option>, <option>x-systemd.wanted-by=</option>,
+        and <option>x-systemd.required-by=</option> is set. See below for detailed information.
+        </para></listitem>
+
+        <listitem><para>Network mount units automatically acquire <varname>After=</varname> dependencies on
+        <filename>remote-fs-pre.target</filename>, <filename>network.target</filename>,
+        plus <varname>After=</varname> and <varname>Wants=</varname> dependencies on <filename>network-online.target</filename>,
+        and a <varname>Before=</varname> dependency on <filename>remote-fs.target</filename>, unless
+        one or more mount options among <option>nofail</option>, <option>x-systemd.wanted-by=</option>,
+        and <option>x-systemd.required-by=</option> is set.</para></listitem>
```

Comment:
Out of scope of this PR, and not your fault, but this section does not mention about the default dependency for tmpfs mount.